32位机器上使用atomic需要保证字节对齐

在arm 32位机器上运行，发现存在字节不对齐问题 [atomic问题](https://pkg.go.dev/sync/atomic#pkg-note-BUG)

panic: unaligned 64-bit atomic operation

goroutine 102 [running]:
runtime/internal/atomic.panicUnaligned()
 /home/go/src/runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Load64(0x293ec34, 0x0, 0x3e8)
 /home/go/src/runtime/internal/atomic/asm_arm.s:263 +0x14
github.com/33cn/chain33/blockchain.(*BlockChain).getMaxSerialChunkNum(...)
 /home/mygo/pkg/mod/github.com/33cn/chain33@v1.67.3/blockchain/chunkshard.go:203
github.com/33cn/chain33/blockchain.(*BlockChain).CheckGenChunkNum(0x293eb00)
 /home/mygo/pkg/mod/github.com/33cn/chain33@v1.67.3/blockchain/chunkshard.go:76 +0x134
github.com/33cn/chain33/blockchain.(*BlockChain).chunkGenerateRoutine(0x293eb00)
 /home/mygo/pkg/mod/github.com/33cn/chain33@v1.67.3/blockchain/chunkshard.go:62 +0x80
created by github.com/33cn/chain33/blockchain.(*BlockChain).InitBlockChain
 /home/mygo/pkg/mod/github.com/33cn/chain33@v1.67.3/blockchain/chain.go:353 +0x290

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -96,6 +96,8 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - uses: actions/checkout@v2
+
       - name: Set up Go
         uses: actions/setup-go@v2
         with:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -91,5 +91,23 @@ jobs:
           make autotest dapp=all
           make docker-compose && make docker-compose-down
 
+  test-goarch-386:
+    name: Run test cases with GOARCH=386
+    runs-on: ubuntu-latest
 
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Build
+        run: make build
+        env:
+          GOARCH: 386
+
+      - name: Run test in 386
+        run: go test ./... -covermode=atomic
+        env:
+          GOARCH: 386
 

--- a/blockchain/chain.go
+++ b/blockchain/chain.go
@@ -64,8 +64,6 @@ type BlockChain struct {
 	//记录本节点已经同步的block高度,用于节点追赶active链,处理节点分叉不同步的场景
 	synBlockHeight int64
 
-	forkPointChan chan int64
-
 	//记录peer的最新block高度,用于节点追赶active链
 	peerList PeerInfoList
 	recvwg   *sync.WaitGroup
@@ -150,6 +148,7 @@ type BlockChain struct {
 
 	// 是否正在下载chunk
 	chunkDownloading int32
+	forkPointChan    chan int64
 }
 
 //New new

--- a/common/common_test.go
+++ b/common/common_test.go
@@ -18,9 +18,9 @@ func TestRandString(t *testing.T) {
 }
 
 func TestInt64Add(t *testing.T) {
-	i := math.MaxInt64
-	assert.Equal(t, i+1, -9223372036854775808)
-	assert.Equal(t, i+2, -9223372036854775807)
+	i := int64(math.MaxInt64)
+	assert.Equal(t, i+1, int64(-9223372036854775808))
+	assert.Equal(t, i+2, int64(-9223372036854775807))
 }
 
 func TestPointer(t *testing.T) {

--- a/system/p2p/dht/protocol/broadcast/broadcast.go
+++ b/system/p2p/dht/protocol/broadcast/broadcast.go
@@ -37,8 +37,8 @@ type broadcastProtocol struct {
 	blockFilter *utils.Filterdata
 	cfg         p2pty.BroadcastConfig
 	ps          *pubsub.PubSub
-	syncStatus  bool
 	currHeight  int64
+	syncStatus  bool
 	lock        sync.RWMutex
 	ltB         *ltBroadcast
 	val         *validator

--- a/system/p2p/dht/protocol/p2pstore/p2pstore.go
+++ b/system/p2p/dht/protocol/p2pstore/p2pstore.go
@@ -38,6 +38,7 @@ var log = log15.New("module", "protocol.p2pstore")
 
 //Protocol ...
 type Protocol struct {
+	concurrency      int64
 	*protocol.P2PEnv //协议共享接口变量
 
 	chunkToSync     chan *types.ChunkInfoMsg
@@ -53,8 +54,6 @@ type Protocol struct {
 	//本节点保存的chunk的索引表，会随着网络拓扑结构的变化而变化
 	localChunkInfo      map[string]LocalChunkInfo
 	localChunkInfoMutex sync.RWMutex
-
-	concurrency int64
 
 	// 扩展路由表，每小时更新一次
 	extendRoutingTable *kbt.RoutingTable

